src/htm/algorithms/SpatialPooler.hpp

```diff
@@ -233,8 +233,15 @@ class SpatialPooler : public Serializable
   @param active An SDR representing the winning columns after
         inhibition. The size of the SDR is equal to the number of
         columns (also returned by the method getNumColumns).
+
+  @param  spatialAnomalyInputValue (optional) `Real` used for computing "spatial anomaly", see @ref `this.spatial_anomaly.compute()`,
+          obtained by @ref `SP.anomaly`
+
    */
-  virtual void compute(const SDR &input, const bool learn, SDR &active);
+  virtual void compute(const SDR &input, 
+		       const bool learn, 
+		       SDR &active, 
+		       const Real spatialAnomalyInputValue = std::numeric_limits<Real>::min());
```

Comment:
had to extend SP.compute() with the extra, optional field. It works ok, but I don't like it too much, maybe better is the proposal with "SP carrying orig input value"?

Reply:
Maybe the anomaly code could be built into the encoder, since the encoder already has access to the original input value?

Alternatively, this piece of code (spatial anomaly detection) could live in its own class?

Reply:
> anomaly code could be built into the encoder, since the encoder already has access to the original input value?

makes sense! This is really more a function of an encoder (input). As I was saying, I don't really like the current implementation, but it is what's used in NAB. 

> spatial anomaly detection) could live in its own class?

later I'm going to implement SP's anomaly detection based on synapses  of SP, so 
- a part of this PR was preparation of the "infrastructure" for the SP.anomaly, later I'd just change the actual method
- you're right this belongs more to the BaseEncoder than to SP

Which way would you suggest, I don't really have a preference now